### PR TITLE
Hooks release

### PR DESCRIPTION
**Pull request by k-kryukov**
Made single folder for all source files of hooks
Included hook tests which allows to test basic usage of hooks
Added PEP8 support and changed hooks in order to meet PEP8 requirements

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,24 +4,51 @@ execute enable_precommits.sh from root repo folder
...
+    - "FIX"...
+    If changes are not big, you can use FIX prefix in commit message
 
 2. commit-msg
-Allows to check whether commit message matches the template "STOR-***** ..." or "FIX ..."
+Allows to check whether commit message matches the templates described in commit-msg-config
 
 3. pre-commit
 Executes couple of utilities which search for mistakes in your code
+!!! (UTILITIES ARE EXECUTED ONLY FOR FILE IN INDEX)
```
Comment (alexeyandriyash) [inline README.md:26]: Is it true? May be for every file was changed?

File: git_hooks_source_files/commit-msg.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+            match = True
+            break
+
+    if not match:
+        warning_text = open(warning_text_path).read()
+        help_text = open(help_text_path).read()
+        print(yellow_color + warning_text + color_off)
+        print(green_color + "HINT: " + color_off + help_text)
+    else:  # MATCH
+        print(green_color + "Good Commit!" + color_off)
```
Comment (alexeyandriyash) [inline git_hooks_source_files/commit-msg.py:46]: Let's make it silent if everything is perfect

File: git_hooks_source_files/hooks_tests/hooks_tests.sh [hunk 3]
```diff
@@ -0,0 +1,57 @@
+#!/bin/bash
+
+# file for testing basic usage of git hooks
+
+file_path=".git/hooks_tests/testHooksLog"
```
Comment (alexeyandriyash) [inline git_hooks_source_files/hooks_tests/hooks_tests.sh:5]: could you add this file to .gitignore please?

File: git_hooks_source_files/hooks_tests/hooks_tests.sh [hunk 4]
```diff
@@ -0,0 +1,57 @@
+#!/bin/bash
+
```
Comment (alexeyandriyash) [inline git_hooks_source_files/hooks_tests/hooks_tests.sh:2]: `set -e` to stop execution if something was failded

File: git_hooks_source_files/installator.py [hunk 5]
```diff
@@ -2,20 +2,24 @@
...
-#def prepare_environment():
+# def prepare_environment():
 #    print(sb.check_output(''))
 
+
 def move_files(path):
     path_to_hooks = path + "/.git/hooks"
 
     os.system(f'cp -r {path}/git_hooks_source_files/. {path_to_hooks}')
     os.system(f'cp -r {path}/git_hooks_source_files/.clang-format {path}')
```
Comment (alexeyandriyash) [inline git_hooks_source_files/installator.py:13]: without `-r` flag `hooks_tests` folder wouldn't be copied

File: git_hooks_source_files/installator.py [hunk 6]
```diff
@@ -2,20 +2,24 @@
...
+    os.system(f'rm -rf {path}/.git/hooks_tests')
+    os.system(f'mv -v {path}/.git/hooks/hooks_tests {path}/.git')
+
 
 def main():
-    path_to_repo= sb.check_output(['pwd']).decode('UTF-8').strip()
-    print(path_to_repo)
+    path_to_repo = sb.check_output(['pwd']).decode('UTF-8').strip()
```
Comment (alexeyandriyash) [inline git_hooks_source_files/installator.py:19]: To get the current working directory you could use
```
import os
cwd = os.getcwd()
```

File: git_hooks_source_files/pre-commit.py [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+    dot_positions_list = []
+
+    for i in range(len(s)):
+        if s[i] == '.':
+            dot_positions_list.append(i)
+    return dot_positions_list
+
+
+def get_index_files():
+    os.system('git status --short > .git/hooks/current_state_of_index')
```
Comment (alexeyandriyash) [inline git_hooks_source_files/pre-commit.py:18]: for a future needs https://gitpython.readthedocs.io/en/stable/tutorial.html

File: git_hooks_source_files/prepare-commit-msg.py [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+        sys.stdout = open(".git/hooks_tests/testHooksLog", "a")
+
+    commit_msg_filepath = sys.argv[1]
+    try:
+        branch = check_output(["git",
+                               "symbolic-ref",
+                               "--short",
+                               "HEAD"]).decode("utf-8").strip()
+    except Exception:
+        branch = "1234"
```
Comment (alexeyandriyash) [inline git_hooks_source_files/prepare-commit-msg.py:36]: You should somehow inform that you wasn't able to get branch name.
Does this script execution have any sense after this exception?

**Comment by alexeyandriyash** [comment]
изображение

**Comment by alexeyandriyash** [comment]
Some small things should be fixed

File: README.md [hunk 9]
```diff
@@ -4,24 +4,51 @@ execute enable_precommits.sh from root repo folder
 Activation/deactivation:
```
[the remark this answers is not in the document]
Comment (k-kryukov) [reply README.md:4]: done

File: git_hooks_source_files/commit-msg.py (hunk 2 above)
Comment (k-kryukov) [reply git_hooks_source_files/commit-msg.py:46]: done

File: git_hooks_source_files/hooks_tests/hooks_tests.sh (hunk 3 above)
Comment (k-kryukov) [reply git_hooks_source_files/hooks_tests/hooks_tests.sh:5]: done

File: README.md (hunk 1 above)
Comment (k-kryukov) [reply README.md:26]: RFC

File: git_hooks_source_files/hooks_tests/hooks_tests.sh (hunk 4 above)
Comment (k-kryukov) [reply git_hooks_source_files/hooks_tests/hooks_tests.sh:2]: done

File: git_hooks_source_files/installator.py (hunk 5 above)
Comment (k-kryukov) [reply git_hooks_source_files/installator.py:13]: yeah, such as pre-commit-message/pre-commit-message-helper and other info files which contains messages given to user
hooks_tests is a single folder which shouldn't be copied

**Comment by TheSalvator** [comment]
.clang-format не нужен, уже есть в проекте

File: git_hooks_source_files/commit-msg [hunk 10]
```diff
@@ -30,9 +36,7 @@ with open(commit_msg_filepath, "r+") as file:
             break
 
     if not match:
-        warning_text = open(".git/hooks/pre-commit-message/pre-commit-message-warnings").read()
-        help_text = open(".git/hooks/pre-commit-message/pre-commit-message-helper").read()
+        warning_text = open(warning_text_path).read()
```
Comment (TheSalvator) [inline git_hooks_source_files/commit-msg:39]: Добавь контекстный менеджер

File: git_hooks_source_files/pre-commit [hunk 11]
```diff
@@ -3,59 +3,66 @@
...
 import os
 import json
+import sys
 
-def dot_pos(s : str):
+
+def dot_pos(s: str):
     dot_positions_list = []
 
     for i in range(len(s)):
```
Comment (TheSalvator) [inline git_hooks_source_files/pre-commit:12]: for index, letter in enumerate(s): ...

File: git_hooks_source_files/pre-commit [hunk 12]
```diff
@@ -3,59 +3,66 @@
...
         if s[i] == '.':
             dot_positions_list.append(i)
     return dot_positions_list
 
+
 def get_index_files():
-    #os.chdir('../../')
     os.system('git status --short > .git/hooks/current_state_of_index')
     f = open(".git/hooks/current_state_of_index", 'r')
```
Comment (TheSalvator) [inline git_hooks_source_files/pre-commit:20]: контекстный менеджер

File: git_hooks_source_files/pre-commit [hunk 13]
```diff
@@ -3,59 +3,66 @@
...
-    keys_as_str = " ".join(keys)
 
+def execute_utility(command: str, files: list):
     for file in files:
-        os.system(f'{utility} {keys_as_str} {file}')
+        os.system(command.format(file=('"' + file + '"')))
+
 
+def main():
+    text_json = open(".git/hooks/utilities-config.json")
```
Comment (TheSalvator) [inline git_hooks_source_files/pre-commit:51]: with open ...

File: git_hooks_source_files/prepare-commit-msg [hunk 14]
```diff
@@ -5,84 +5,110 @@ import sys
...
-            branch_pattern_pos = re.search(all_regex[pattern_name], branch)
-            break
+    commit_msg_cfg_path = ".git/hooks/commit_msg_config.json"
+    warning_msg_path = ".git/hooks/prepare-commit-message/warning"
+    hook_name = 'prepare_commit_msg'
 
-    prev_msg_content = f.readlines()
-    not_comment_lines = ["" if cur_s.startswith('#') else cur_s for cur_s in prev_msg_content] # worthful part of message
-    not_comment_msg_text = "".join(not_comment_lines)
+    commit_msg_config = json.load(open(commit_msg_cfg_path))[hook_name]
```
Comment (TheSalvator) [inline git_hooks_source_files/prepare-commit-msg:41]: with open ...

**Review by TheSalvator — CHANGES_REQUESTED** [review]

File: git_hooks_source_files/commit-msg (hunk 10 above)
Comment (k-kryukov) [reply git_hooks_source_files/commit-msg:39]: done

**Pull request merged**